internal/db/types.go

```diff
@@ -0,0 +1,156 @@
+package db
+
+import (
+	"context"
+
+	"github.com/redhatinsights/platform-changelog-go/internal/structs"
+)
+
+type DBConnector interface {
```

Comment:
I would consider breaking this interface into multiple smaller interfaces.

There already appears to be a logical separation (Commits, Servcies, Timelines, Deployments).  I would try splitting this big interface into smaller type specific interfaces:  CommitRespository, ServiceRespository, TimelineRepository, DeploymentRespository.  Something like that.

I would also consider removing the `structs` package (in the future...not necessary in this PR).  The structs in the `structs` package belong close to where they are actually being used.  

I would also try to avoid having the database layer "know" how to generate the json that belongs in the http layer.  This means duplicating the structs/data at the http layer and database layer.

Yes, this type of duplication is gross, but I tend to think this is ok (but it is tradeoff to think about).  It is my experience that a lot of time, you end up needing multiple different views of the same data in the http layer.  The database layer should not know how the http layer needs to format the outgoing data.  This json data formatting is the responsibility of the http layer.  The "gross" duplication will allow the http layer and database layer to vary independently.  Just something to think about for the future!  :)

Sorry for writing a book...   :(

Reply:
After looking a little deeper, it appears there isn't a clear separation in the DB between commits, services, timelines, and deployments. Commits and deployments source their data from the timelines table, and the expanded service data also requires the timelines table. 

For now, because of the cross-access of these tables, I think it would be best to keep the data together in one interface that has access to the timelines and services tables. At a later time, It could be a good idea to restructure this so that the splitting of this information happens at the API layer instead of here where the DB is accessed. That way, we could break apart the logic to contain only a `ServiceRepository` and `TimelineRepository` (matching the structure of the DB) and the Services, Timelines, Commits, Deploys, and ExpandedService would be split out by the API controllers.